unblob/handlers/archive/autel/ecc.py

```diff
@@ -0,0 +1,333 @@
+    (224, 0),
+    (27, 166),
+    (142, 0),
+    (27, 192),
+    (24, 132),
+    (12, 183),
+    (96, 133),
+    (84, 192),
+    (14, 132),
+    (27, 178),
+    (10, 140),
+    (155, 0),
+    (9, 132),
+    (17, 160),
+    (56, 133),
+    (96, 192),
+    (82, 132),
+    (13, 160),
+    (27, 137),
+    (20, 193),
+    (139, 0),
+    (28, 161),
+    (145, 0),
+    (19, 192),
+    (118, 132),
+    (115, 165),
+    (20, 132),
+    (145, 0),
+    (14, 132),
+    (12, 167),
+    (146, 0),
+    (17, 193),
+    (29, 132),
+    (96, 176),
+    (28, 144),
+    (27, 193),
+    (140, 0),
+    (31, 180),
+    (148, 0),
+    (27, 192),
+    (14, 132),
+    (83, 160),
+    (18, 137),
+    (17, 193),
+    (23, 132),
+    (13, 165),
+    (13, 145),
+    (151, 0),
+    (147, 0),
+    (27, 178),
+    (96, 137),
+    (19, 193),
+    (159, 0),
+    (14, 160),
+    (25, 148),
+    (17, 193),
+    (142, 0),
+    (16, 180),
+    (27, 136),
+    (14, 193),
+    (224, 0),
+    (17, 178),
+    (12, 144),
+    (224, 0),
+    (28, 132),
+    (27, 160),
+    (13, 141),
+    (11, 193),
+    (96, 132),
+    (27, 165),
+    (30, 140),
+    (224, 0),
+    (146, 0),
+    (31, 165),
+    (29, 129),
+    (96, 192),
+    (140, 0),
+    (31, 161),
+    (24, 145),
+    (140, 0),
+    (96, 132),
+    (27, 165),
+    (29, 140),
+    (31, 192),
+    (154, 0),
+    (14, 161),
+    (27, 145),
+    (140, 0),
+    (18, 132),
+    (23, 167),
+    (96, 140),
+    (21, 129),
+    (14, 132),
+    (17, 165),
+    (9, 137),
+    (12, 193),
+    (155, 0),
+    (18, 161),
+    (96, 141),
+    (27, 192),
+    (148, 0),
+    (29, 178),
+    (23, 133),
+    (24, 192),
+    (155, 0),
+    (10, 180),
+    (96, 133),
+    (28, 192),
+    (14, 132),
+    (31, 130),
+    (28, 129),
+    (18, 193),
+    (31, 132),
+    (12, 180),
+    (13, 144),
+    (96, 193),
+    (31, 132),
+    (96, 160),
+    (13, 141),
+    (27, 193),
+    (18, 132),
+    (23, 181),
+    (26, 140),
+    (27, 193),
+    (156, 0),
+    (96, 166),
+    (79, 141),
+    (211, 0),
+    (76, 132),
+    (77, 160),
+    (75, 133),
+    (206, 0),
+    (182, 0),
+    (96, 129),
+    (59, 133),
+    (191, 0),
+    (173, 0),
+]
+
+
+def ecc_decode(data: bytes):
+    """Decode a 256 byte data block."""
+    assert len(data) <= 256
+    for i, value in enumerate(data):
+        a, b = KEYS[i]
+        yield ((value + a) ^ b) % 256
+
+
+class ECCExtractor(Extractor):
+    def __init__(self):
+        self._struct_parser = StructParser(C_DEFINITIONS)
+
+    def extract(self, inpath: Path, outdir: Path):
+        outpath = outdir.joinpath(f"{inpath.name}.decrypted")
+        with File.from_path(inpath) as file:
+            header = self._struct_parser.parse(
+                "autel_header_t", file, endian=Endian.LITTLE
+            )
+            outfile = outpath.open("wb")
+            for data in iterate_file(file, header.header_size, header.file_size, 256):
+                outfile.write(bytes(ecc_decode(data)))
+        outfile.close()
```

Comment:
Opening-closing of `outfile` is a bit uneven.
```suggestion
            with outpath.open("wb") as outfile:
                for data in iterate_file(file, header.header_size, header.file_size, 256):
                    outfile.write(bytes(ecc_decode(data)))
```